pipeline/transformer/transformer_calm/src/main/scala/uk/ac/wellcome/platform/transformer/calm/CalmTransformer.scala

```diff
@@ -1,28 +1,174 @@
 package uk.ac.wellcome.platform.transformer.calm
 
-import uk.ac.wellcome.models.work.internal.{
-  SourceIdentifier,
-  TransformedBaseWork,
-  UnidentifiedWork,
-  WorkData
-}
+import uk.ac.wellcome.models.work.internal._
 import uk.ac.wellcome.platform.transformer.calm.models.CalmRecord
+import uk.ac.wellcome.models.work.internal.result._
+
+object CalmTransformer extends Transformer[CalmRecord] with CalmOps {
+
+  val identifierMapping = Map(
+    "RefNo" -> CalmIdentifierTypes.refNo,
+    "AltRefNo" -> CalmIdentifierTypes.altRefNo,
+    "BNumber" -> IdentifierType("sierra-system-number")
+  )
 
-trait CalmTransformerError extends Throwable
-case object SourceIdentifierMissingError extends CalmTransformerError
-case object MultipleSourceIdentifiersFoundError extends CalmTransformerError
+  val notesMapping = List(
+    ("AdminHistory", BiographicalNote(_)),
+    ("CustodHistory", OwnershipNote(_)),
+    ("Acquisition", AcquisitionNote(_)),
+    ("Appraisal", AppraisalNote(_)),
+    ("Accruals", AccrualsNote(_)),
+    ("RelatedMaterial", RelatedMaterial(_)),
+    ("PubInNote", PublicationsNote(_)),
+    ("UserWrapped4", FindingAids(_)),
+    ("Copyright", CopyrightNote(_)),
+    ("ReproductionConditions", TermsOfUse(_)),
+    ("Arrangement", ArrangementNote(_))
+  )
 
-object CalmTransformer extends Transformer[CalmRecord] {
-  def transform(
-    record: CalmRecord): Either[CalmTransformerError, TransformedBaseWork] =
-    Right(
+  def apply(record: CalmRecord): Result[TransformedBaseWork] =
+    workData(record) map { data =>
       UnidentifiedWork(
-        sourceIdentifier = SourceIdentifier(
-          value = record.id,
-          identifierType = CalmIdentifierTypes.recordId,
-          ontologyType = "IdentifierType"),
-        version = 1,
-        data = WorkData()
+        sourceIdentifier = sourceIdentifier(record),
+        version = 0,
```

Comment:
Is this how all the transformers work?

Reply:
Nope (if you mean setting `version = 0`), other transformers initialise this to the source version, but we don't have a version field returned from Calm.

Reply:
I lied! It actually uses the adapter version, good thing you made me check